app/controllers/subscriptions_management_controller.rb

```diff
@@ -0,0 +1,125 @@
+class SubscriptionsManagementController < ApplicationController
+  before_action :set_cache_control_header
+  before_action :check_if_access_is_enabled
+  before_action :require_authentication
+  before_action :get_subscription_details
+  before_action :set_back_url
+
+  MISSING_EMAIL_ERROR = 'Please enter your email address.'.freeze
+  INVALID_EMAIL_ERROR = 'That email address isn’t valid, or it’s already in use – check you’ve typed in your email address correctly.'.freeze
+
+  def index; end
+
+  def update_frequency
+    @subscription_id = subscription_id
+    @current_frequency = @subscriptions[@subscription_id]['frequency']
+  end
+
+  def change_frequency
+    email_alert_api.change_subscription(
+      id: subscription_id,
+      frequency: new_frequency
+    )
+
+    redirect_to list_subscriptions_path
+  end
+
+  def update_address
+    @address = @subscriber['address']
+  end
+
+  def change_address
+    @address = @subscriber['address']
+
+    unless subscriber_params[:new_address].present?
+      flash.now[:error] = MISSING_EMAIL_ERROR
+      render :update_address
+      return
+    end
+
+    begin
+      email_alert_api.change_subscriber(
+        id: subscriber_id,
+        new_address: new_address
+      )
+    rescue GdsApi::HTTPUnprocessableEntity
+      @new_address = new_address
+      flash.now[:error] = INVALID_EMAIL_ERROR
+      render :update_address
+      return
+    end
+
+    redirect_to list_subscriptions_path
+  end
+
+  def confirm_unsubscribe_all; end
+
+  def confirmed_unsubscribe_all
+    begin
+      email_alert_api.unsubscribe_subscriber(subscriber_id)
+    rescue GdsApi::HTTPNotFound
+      # The user has already unsubscribed.
+      nil
+    end
+  end
+
+private
+
+  def check_if_access_is_enabled
+    # TODO: Remove this once we're ready to go live
+    render plain: '404 Not Found', status: 404 unless ENV['SUBSCRIPTION_MANAGEMENT_ENABLED']
+  end
+
+  def set_cache_control_header
+    headers['Cache-Control'] = 'private'
+  end
```

Comment:
If we're going to be using this a lot I feel like it should be in the application controller - could even be a default action that you skip if you don't want it.

Reply:
I've moved this into the application controller since it makes sense to me that none of the subscribe/unsubscribe/manage journey should really be cached.